cmd/object-api-utils.go

```diff
@@ -302,3 +304,30 @@ type byBucketName []BucketInfo
 func (d byBucketName) Len() int           { return len(d) }
 func (d byBucketName) Swap(i, j int)      { d[i], d[j] = d[j], d[i] }
 func (d byBucketName) Less(i, j int) bool { return d[i].Name < d[j].Name }
+
+// cleanMinioInternalMetadataKeys removes X-Amz-Meta- prefix from minio internal
+// encryption metadata that was sent by minio gateway
+func cleanMinioInternalMetadataKeys(metadata map[string]string) map[string]string {
+	var newMeta = make(map[string]string)
+	for k, v := range metadata {
+		if strings.HasPrefix(k, "X-Amz-Meta-X-Minio-Internal-") {
+			newMeta[strings.TrimPrefix(k, "X-Amz-Meta-")] = v
+		} else {
+			newMeta[k] = v
+		}
+	}
+	return newMeta
+}
+
+// returns true if object is sse encrypted at the gateway
+func isGatewayEncrypt(metadata map[string]string, headers http.Header) bool {
+	if GlobalGatewaySSE != nil && (hasServerSideEncryptionHeader(headers) || crypto.SSECopy.IsRequested(headers)) {
```

Comment:
Why is `crypto.SSECopy.IsRequested(headers))` necessary? From my understanding the function returns true if
the object is encrypted. This does/should not depend whether the client sends SSE-C/SSE-C copy headers.

Reply:
this was not necessary and has been removed.